radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,293 @@
+/**
+ * @file RAntenna.c
+ * @date Jan 15, 2022
+ * @author Addi Amaya (caa746)
+ */
+
+#include <RAntenna.h>
+#include <satellite-subsystems/IsisAntS.h>
+#include <string.h>
+#include <hal/errors.h>
+#include <hal/Timing/Time.h>
+
+/***************************************************************************************************
+                                  PRIVATE DEFINITIONS AND VARIABLES
+***************************************************************************************************/
+
+/** Struct for defining Antenna Systems I2C Addresses. Configured as a single bus, both address write to same location */
+ISISantsI2Caddress RAntennaI2Caddress = {ANTENNA_I2C_SLAVE_ADDR_PRIMARY,ANTENNA_I2C_SLAVE_ADDR_REDUNANT};
+
+/** Struct that holds the current deployment status of the antenna*/
+antenna_deployment_status_t RAntennaStatus;
+
+/** Track whether the antenna has been initialized yet */
+static int antennaInitialized = 0;
+
+/** Track Number of times the antenna attempted to deploy*/
+static int antennaDeploymentAttempts = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Initializes the ISISpace Antenna driver
+ *
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaInit(void) {
+
+	// Only allow initialization once (return without error if already initialized)
+	if (antennaInitialized)
+		return 0;
+
+	// Isis Function call for antenna initialization
+	int error = IsisAntS_initialize(&RAntennaI2Caddress, ANTENNAS_ON_BOARD);
+
+	// Update flag if successfully initialized
+	if (!error)
+		antennaInitialized = 1;
+
+	// TODO: record errors (if present) to System Manager
```

Comment:
I think we should change this to 1 for failure and 0 for success like the rest of our stuff. If/when we implement an error manager this would be a hardware level issues and would be handled by that.

Reply:
I understand what you're saying but would it not be a better approach to have a Rcommon.h (something that is already in Ty's main clean up branch) and set a define as `#define INITIALIZED 1` to make it easier to understand. Additionally, I think that having the variable antennaInitialized set to 1 on success makes more sense to me. If it is detrimental to maintain this, I can easily fix it later once we have further defined the error handler